### PR TITLE
feat(releases): Add endpoint for project release setup status

### PR DESCRIPTION
**Pull request by denamwangi**
This sets up the first base for updates to the releases tab. This returns info on what steps the project has completed so far. See spec [here](https://paper.dropbox.com/doc/Spec-Releases-tab--ANo1zkOS8drR7JA7ey_QLKKMAg-Ie1TFu4Vn1znYvYwlVp3G)

### CODE REVIEW

File: src/sentry/api/endpoints/project_release_setup.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+from sentry.models import Group, ReleaseCommit, Repository, Deploy
+from rest_framework.response import Response
+
+
+class ProjectReleaseSetupEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectReleasePermission, )
+
+    def get(self, request, project):
+        """
+        Checks how set up a project is on releases.
```
Comment (mnoble) [inline src/sentry/api/endpoints/project_release_setup.py]: I don't understand what this says, haha.

**Review by mnoble — COMMENTED** [review]
How is this endpoint being consumed? It's a bit odd to me to have an endpoint that only checks the existence of other resources.

File: src/sentry/api/endpoints/project_release_setup.py (hunk 1 above)
Comment (denamwangi) [reply src/sentry/api/endpoints/project_release_setup.py]: Lol fair- i'll make it clearer. It's meant to see what steps have been completed of the 4 required to be "fully" set up on releases.

**Comment by denamwangi** [comment]
So this will be consumed on the front end on our releases tab to help inform users what steps they have yet to complete to finish setting up releases. All our existing endpoints need some level of gymnastics to get at the same level of info - def open to suggestions here if there's a better way

File: src/sentry/api/endpoints/project_release_setup.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/endpoints/project_release_setup.py]: Could use `Get release setup progress for a project`.

**Comment by mnoble** [comment]
What if we framed this in a more resourceful way, as something like `ProjectReleaseSetupStatus` or `ProjectReleaseSetupCompletion`?

We could expose it as something like:

```json
[
  {
    "step": "tag",
    "complete": true
  },
  {
    "step": "commit",
    "complete": true
  },
  {
    "step": "repo",
    "complete": true
  },
  {
    "step": "deploy",
    "complete": true
  }
]
```

It's a little more structured this way and leaves the door open for adding additional data to each object (contrived example: we add completion % to each step), if we ever wanted to down the line, without making backwards incompatible changes to the API.

I personally find it easier to reason with API endpoints when they represent and expose some sort of domain object.

Thoughts, @markstory?

**Comment by markstory** [comment]
@mnoble I generally prefer to have APIs respond with 'resources' as well. Like you said it makes future expansion easier, and follows patterns we use elsewhere in the API.

**Comment by denamwangi** [comment]
Excellent points thanks. This all makes sense - i'll update

File: src/sentry/api/urls.py [hunk 2]
```diff
@@ -812,9 +812,9 @@
         name='sentry-api-0-project-releases-token'
     ),
     url(
-        r'^projects/(?P<organization_slug>[^\/]+)/(?P<project_slug>[^\/]+)/releases/status/$',
-        ProjectReleaseSetupEndpoint.as_view(),
-        name='sentry-api-0-project-releases-status'
+        r'^projects/(?P<organization_slug>[^\/]+)/(?P<project_slug>[^\/]+)/releases/completion/$',
```
[the remark this answers is not in the document]
Comment (denamwangi) [reply src/sentry/api/urls.py]: Good point-just [checked](https://redash.getsentry.net/queries/1346/source) and currently no users have a release of `token` or `completion`. Def +1 on reserving these at some point (if not a blocker for this)

**Review by mnoble — APPROVED** [review]
Good stuff! 👍

**Pull request merged**